File: _includes/code/java-v6/src/test/java/QuickstartLocalTest.java
import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.CollectionHandle;
import io.weaviate.client6.v1.api.collections.Generative;
import io.weaviate.client6.v1.api.collections.Property;
import io.weaviate.client6.v1.api.collections.VectorConfig;
import io.weaviate.client6.v1.api.collections.data.InsertManyResponse;

import org.json.JSONArray;
import org.json.JSONObject;
import org.junit.jupiter.api.Test;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class QuickstartLocalTest {

  @Test
  void testConnectionIsReady() throws Exception {
    // START InstantiationExample
    WeaviateClient client = WeaviateClient.connectToLocal();
    // highlight-start
    System.out.println(client.isReady()); // Should print: `True`
    // highlight-end

    client.close(); // Free up resources

    // END InstantiationExample
  }

  @Test
  void testCreateCollection() throws Exception {
    String collectionName = "Question";
    // START CreateCollection
    WeaviateClient client = WeaviateClient.connectToLocal();

    // highlight-start
    client.collections.create(
        collectionName,
        col -> col
            .vectorConfig(VectorConfig.text2vecTransformers()) // Configure the Weaviate Embeddings integration
            .generativeModule(Generative.cohere()) // Configure the Cohere generative AI integration
            .properties(Property.text("answer"), Property.text("question"), Property.text("category")));
    CollectionHandle<Map<String, Object>> questions = client.collections.use(collectionName);
    // highlight-end
    // END CreateCollection
    client.collections.delete(collectionName);
    // START CreateCollection

    client.close(); // Free up resources

    // END CreateCollection
  }

  @Test
  void testImportDataWorkflow() throws Exception {
    // START Import
    WeaviateClient client = WeaviateClient.connectToLocal();

    // Create the collection
    String collectionName = "Question";
    client.collections.create(collectionName, col -> col
        .properties(
            Property.text("answer"),
            Property.text("question"),
            Property.text("category"))
        .vectorConfig(VectorConfig.text2vecTransformers())); // Configure the Weaviate Embeddings integration;

    // Get JSON data using HttpURLConnection
    URL url = new URL("https://raw.githubusercontent.com/weaviate-tutorials/quickstart/main/data/jeopardy_tiny.json");
    String jsonData = new BufferedReader(
        new InputStreamReader(((HttpURLConnection) url.openConnection()).getInputStream()))
        .lines().reduce("", String::concat);

    // highlight-start
    CollectionHandle<Map<String, Object>> questions = client.collections.use(collectionName);
    List<Map<String, Object>> questionsToInsert = new ArrayList<>();

    // Parse and prepare objects using org.json
    new JSONArray(jsonData).forEach(item -> {
      JSONObject json = (JSONObject) item;
      Map<String, Object> properties = new HashMap<>();
      properties.put("answer", json.getString("Answer"));
      properties.put("question", json.getString("Question"));
      properties.put("category", json.getString("Category"));
      questionsToInsert.add(properties);
    });

    // Call insertMany with the list of objects
    InsertManyResponse insertResponse = questions.data.insertMany(questionsToInsert.toArray(new Map[0]));
    // highlight-end

    // Check for errors
    if (!insertResponse.errors().isEmpty()) {
      System.err.printf("Number of failed imports: %d\n", insertResponse.errors().size());
      System.err.printf("First failed object error: %s\n", insertResponse.errors().get(0));
    } else {
      System.out.printf("Successfully inserted %d objects.\n", insertResponse.uuids().size());
    }
    // END Import
    // client.collections.delete(collectionName);
    // START Import

    client.close(); // Free up resources
    // END Import
  }

  @Test
  void testNearTextQuery() throws Exception {
    // START NearText
    WeaviateClient client = WeaviateClient.connectToLocal();

    String collectionName = "Question";
    var questions = client.collections.use(collectionName);

    // highlight-start
    var response = questions.query.nearText("biology", q -> q.limit(2));
    // highlight-end

    for (var obj : response.objects()) {
      System.out.println(obj.properties());
    }
    // END NearText
    client.collections.delete(collectionName);
    // START NearText

    client.close(); // Free up resources
    // END NearText
  }

  // @Test
  // void testRagQuery() {
  // WeaviateClient client = WeaviateClient.connectToLocal();

  // var questions = client.collections.use("Question");

  // // highlight-start
  // var response = questions.generate.nearText(
  // q -> q
  // .query("biology")
  // .limit(2),
  // g -> g.groupedTask("Write a tweet with emojis about these facts."));
  // // highlight-end

  // System.out.println(response.generative().text()); // Inspect the generated
  // text
  // }
  // START RAG
  // Coming soon
  // END RAG
}

File: _includes/code/csharp/QuickstartLocalTest.cs
using Weaviate.Client;
using Weaviate.Client.Models;
using System;
using System.Threading.Tasks;
using System.Text.Json;
using System.Linq;
using System.Collections.Generic;
using System.Net.Http;
using Weaviate.Client.Models.Generative;
using Xunit;

namespace WeaviateProject.Examples;

[Collection("Sequential")] // Ensures tests in this class run one after another
public class QuickstartLocalTest
{
    [Fact]
    public async Task TestConnectionIsReady()
    {
        // START InstantiationExample
        using var client = await Connect.Local();

        // highlight-start
        // GetMeta returns server info. A successful call indicates readiness.
        var meta = await client.GetMeta();
        Console.WriteLine(meta);
        // highlight-end

        // The 'using' statement handles freeing up resources automatically.
        // END InstantiationExample
    }

    [Fact]
    public async Task FullQuickstartWorkflowTest()
    {
        using var client = await Connect.Local();
        string collectionName = "Question";

        // Clean up previous runs if they exist
        if (await client.Collections.Exists(collectionName))
        {
            await client.Collections.Delete(collectionName);
        }

        // START CreateCollection
        // highlight-start
        var questions = await client.Collections.Create(new CollectionConfig
        {
            Name = collectionName,
            Properties =
            [
                    Property.Text("answer"),
                    Property.Text("question"),
                    Property.Text("category")
            ],
            VectorConfig = Configure.Vectors.Text2VecTransformers().New("default"),  // Configure the text2vec-transformers integration
            GenerativeConfig = new GenerativeConfig.Cohere() // Configure the Cohere generative AI integration
        });
        // highlight-end
        // END CreateCollection

        // START Import
        // Get JSON data using HttpClient
        using var httpClient = new HttpClient();
        var jsonData = await httpClient.GetStringAsync("https://raw.githubusercontent.com/weaviate-tutorials/quickstart/main/data/jeopardy_tiny.json");

        // highlight-start
        var questionsToInsert = new List<object>();

        // Parse and prepare objects using System.Text.Json
        var jsonObjects = JsonSerializer.Deserialize<List<Dictionary<string, JsonElement>>>(jsonData);
        foreach (var jsonObj in jsonObjects)
        {
            questionsToInsert.Add(new
            {
                answer = jsonObj["Answer"].GetString(),
                question = jsonObj["Question"].GetString(),
                category = jsonObj["Category"].GetString()
            });
        }

        // Call InsertMany with the list of objects converted to an array
        var insertResponse = await questions.Data.InsertMany(questionsToInsert.ToArray());
        // highlight-end
        // END Import

        // Check for errors
        if (insertResponse.HasErrors)
        {
            Console.WriteLine($"Number of failed imports: {insertResponse.Errors.Count()}");
            Console.WriteLine($"First failed object error: {insertResponse.Errors.First()}");
        }
        else
        {
            Console.WriteLine($"Successfully inserted {insertResponse.Objects.Count()} objects.");
        }

        // START NearText
        // highlight-start
        var response = await questions.Query.NearText("biology", limit: 2);
        // highlight-end

        foreach (var obj in response.Objects)
        {
            Console.WriteLine(JsonSerializer.Serialize(obj.Properties));
        }
        // END NearText

        // START RAG
        // highlight-start
        var ragResponse = await questions.Generate.NearText(
            "biology",
            limit: 2,
            groupedTask: new GroupedTask("Write a tweet with emojis about these facts."),
            provider: new Providers.OpenAI {}
        );
        // highlight-end

        // Inspect the results
        Console.WriteLine(JsonSerializer.Serialize(ragResponse.Generative.Values));
        // END RAG
        await client.Collections.Delete(collectionName);
    }
}